### PR TITLE
Add IRCv3 away-notify capability

### PR DESCRIPTION
**Pull request by vanosg**
Patch by: Geo

One-line summary:
Add IRCv3 away-notify capability

Additional description (if needed):
Adds a LOG_SERV message to console/log, and a bind AWY3 to handle away-notify messages. To catch a user returning from an away state, the mask should be set to ""

Test cases demonstrating functionality (if applicable):

Setup:
```
[02:57:06] CAP: Current Negotiations away-notify with irc.foo.net
.tcl proc awaybind {from msg} {putlog "yay! msg is $msg from is $from"}
.tcl bind awy3 * * awaybind
```
User enters away state:
```
[02:57:38] testers!~test@DBA7DD58:81C6A49C:168A19B8:IP is now away: foo
[02:57:38] yay! msg is foo from is testers!~test@DBA7DD58:81C6A49C:168A19B8:IP,
```
(returning from away status)
```
[02:57:41] testers!~test@DBA7DD58:81C6A49C:168A19B8:IP has returned from away status
[02:57:41] yay! msg is  from is testers!~scase@DBA7DD58:81C6A49C:168A19B8:IP,
.binds
[02:58:13] #-HQ# binds 
Command bindings:
  TYPE FLAGS    COMMAND              HITS BINDING (TCL)
  awy3 -|-      *                       2 awaybind
```

### CODE REVIEW

File: doc/sphinx_source/mainDocs/tcl-commands.rst [hunk 1]
```diff
@@ -3311,6 +3311,16 @@ The following is a list of bind types and how they work. Below each bind type is
 
   Module: core
 
+(50) AWY3 (stackable)
```
Comment (thommey) [inline doc/sphinx_source/mainDocs/tcl-commands.rst:3314]: please call this "bind away"

File: doc/sphinx_source/mainDocs/tcl-commands.rst [hunk 2]
```diff
@@ -3311,6 +3311,16 @@ The following is a list of bind types and how they work. Below each bind type is
...
+
+  bind awy3 <flags> <mask> <proc>
+
+  procname <from> <msg>
```
Comment (thommey) [inline doc/sphinx_source/mainDocs/tcl-commands.rst:3318]: arguments should be identical to other binds:
`procname <nick> <host> <hand> <chan> <awaymsg>` (awaymsg is "" if they return)
and call once per channel the user is on

File: src/mod/server.mod/servmsg.c [hunk 3]
```diff
@@ -1412,6 +1431,19 @@ static int handle_sasl_timeout()
   return sasl_error("timeout");
 }
 
+/* Got AWAY message; only valid for IRCv3 away-notify capability */
+static int gotawayv3(char *from, char *msg)
+{
+  if (strlen(msg)) {
+    fixcolon(msg);
+    putlog(LOG_SERV, "*", "%s is now away: %s", from, msg);
```
Comment (thommey) [inline src/mod/server.mod/servmsg.c:1439]: I believe this should be LOG_JOIN | LOG_MISC just like the "%s joined %s" message for joins

File: src/mod/server.mod/servmsg.c [hunk 4]
```diff
@@ -1412,6 +1431,19 @@ static int handle_sasl_timeout()
...
+  } else {
+    putlog(LOG_SERV, "*", "%s has returned from away status", from);
+  }
+  check_tcl_awayv3(from, msg);
```
Comment (thommey) [inline src/mod/server.mod/servmsg.c:1443]: usually, we call binds before logging I believe

File: src/mod/server.mod/servmsg.c [hunk 5]
```diff
@@ -264,6 +264,18 @@ static int check_tcl_wall(char *from, char *msg)
   return 1;
 }
 
+static int check_tcl_awayv3(char *from, char *msg)
+{
+  int x;
+
+  Tcl_SetVar(interp, "_awayv31", from, 0);
+  Tcl_SetVar(interp, "_awayv32", msg ? (char *) msg : "", 0);
```
Comment (thommey) [inline src/mod/server.mod/servmsg.c:272]: that cast isn't necessary, msg is already charptr

File: doc/sphinx_source/mainDocs/tcl-commands.rst (hunk 1 above)
Comment (vanosg) [reply doc/sphinx_source/mainDocs/tcl-commands.rst:3314]: we already have a 'bind away' for partyline away status

File: doc/sphinx_source/mainDocs/tcl-commands.rst (hunk 2 above)
Comment (vanosg) [reply doc/sphinx_source/mainDocs/tcl-commands.rst:3318]:
```
.tcl proc foo {nick user hand chan msg} {putlog "nick: $nick user: $user hand: $hand chan: $chan msg: $msg"}
.tcl bind * * foo

[05:10:08] nick: foo, user: foo!foo@foo.com hand: Foo chan: #eggdroptest msg: :This is a test message
[05:10:08] foo!foo@foo.com is now away: This is a test message
[05:10:12] nick: foo, user: foo!foo@foo.com hand: Foo chan: #eggdroptest msg: 
```
Adding a channel mask, bot is on two channels #eggdroptest and #eggtest:
```
.tcl bind awy3 * "#eggdroptest *" goo
.tcl proc goo {nick user hand chan text} {putlog "This was activated by $chan"}

[05:14:49] This was activated by #eggdroptest
[05:14:49] nick: foo, user: foo!foo@foo.com hand: Foo chan: #eggdroptest msg: :Going away now
[05:14:49] nick: foo, user: foo!foo@foo.com hand: Foo chan: #eggtest msg: :Going away now
```

**Pull request merged**